apps/algorithm/recommend/src/server.py

```diff
@@ -16,20 +18,43 @@
 recommender = Recommender()
 
 
+@app.middleware("http")
+async def authenticate_request(request: Request, call_next):
+    auth_token = request.cookies.get("authorization")
+
+    if not auth_token:
+        raise HTTPException(status_code=401, detail="Authorization header missing")
+
+    if not os.getenv("PYTEST_CURRENT_TEST"):
```

Comment:
This will need some manual testing to ensure that auth doesn't just allow through users when running in prod

Reply:
It works all right